org.eclipse.richbeans.widgets/src/org/eclipse/richbeans/widgets/scalebox/NumberBox.java

```diff
@@ -781,8 +781,10 @@ public void setDecimalPlaces(int decimalPlaces) {
 	 */
 	@SuppressWarnings("unchecked")
 	public double getMaximum() {
-		if (maxProvider != null)
-			return maxProvider.getBoundValue();
+		if (maxProvider != null) {
+			double dbl = maxProvider.getBoundValue();
+			if (!Double.isNaN(dbl)) return dbl;
```

Comment:
dbl is a primitive which cannot be null. There is an event which comes through there the value is NaN during linking the UI. This event should not be sent but was a more risky change :) 

Thanks for checking about IBeanController, I think I will add its removal to this PR.